framework/web/AssetManager.php

```diff
@@ -576,6 +587,10 @@ public function getPublishedUrl($path)
      */
     protected function hash($path)
     {
+        if (!empty($this->hashCallback)) {
+            return call_user_func($this->hashCallback, $path);
+        }
+        $path = (pathinfo($path, PATHINFO_EXTENSION) === '' ? $path : dirname($path)) . filemtime($path);
```

Comment:
@samdark @klimov-paul 

I'm wondering if we should just pass this `$path = (pathinfo($path, PATHINFO_EXTENSION) === '' ? $path : dirname($path))` to the `hashCallback` or let the users recheck the `$path`?